server/channels/api4/user.go

```diff
@@ -2262,6 +2264,38 @@ func attachDeviceId(c *Context, w http.ResponseWriter, r *http.Request) {
 	ReturnStatusOK(w)
 }
 
+func setExtraSessionProps(c *Context, w http.ResponseWriter, r *http.Request) {
+	newProps := map[string]string{}
+	receivedProps := model.MapFromJSON(r.Body)
+
+	deviceNotificationsDisabled := receivedProps[model.SessionPropDeviceNotificationDisabled]
+	if deviceNotificationsDisabled != "" && deviceNotificationsDisabled != "false" && deviceNotificationsDisabled != "true" {
+		c.SetInvalidParam(model.SessionPropDeviceNotificationDisabled)
+		return
```

Comment:
Given that now we have several possible props to update, should we not return here and let it continue evaluating the other ones so we don't lose some extra info

Reply:
My idea is that if you send a prop, the prop must be valid, or error out. Validating all of them seems unnecessary, and adding just part of the information seems dangerous.

Reply:
Fair enough 👍